pkg/client/edv/client.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package edv
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/trustbloc/edv/pkg/restapi/edv/operation"
+)
+
+// Client is used to interact with an EDV server.
+type Client struct {
+	edvServerURL string
+}
+
+// New returns a new instance of an EDV client.
+func New(edvServerURL string) *Client {
+	return &Client{edvServerURL: edvServerURL}
+}
+
+// CreateDataVault sends the EDV server a request to create a new data vault.
+// The location of the newly created data vault is returned.
+func (c *Client) CreateDataVault(config *operation.DataVaultConfiguration) (string, error) {
+	jsonToSend, err := json.Marshal(config)
+	if err != nil {
+		return "", err
+	}
+
+	resp, err := http.Post(c.edvServerURL+"/data-vaults", "application/json", bytes.NewBuffer(jsonToSend))
+	if err != nil {
+		return "", err
+	}
+
+	defer func() {
+		err = resp.Body.Close()
```

Comment:
duplicate code

Reply:
The linter complains that the response body must be closed if I move this code to a new function and call defer on that. Any idea why?

Reply:
I've verified that it does indeed execute the deferred function.

Reply:
nevermind. I found a way to write it so that the linter doesn't complain. It seems to not complain if I wrap the function that does the closing inside of a func() like this:

`defer func() {
		closeBody(resp)
	}()`

Reply:
Double nevermind. Now the linter doesn't give me that original error any more. So ignore this whole chain of comments.